spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/convert/MappingSpannerReadConverter.java

```diff
@@ -46,60 +40,17 @@
 class MappingSpannerReadConverter extends AbstractSpannerCustomConverter
 		implements SpannerEntityReader {
 
-	// @formatter:off
-	static final Map<Class, BiFunction<Struct, String, List>> readIterableMapping =
-			new ImmutableMap.Builder<Class, BiFunction<Struct, String, List>>()
-					// @formatter:on
-					.put(Boolean.class, AbstractStructReader::getBooleanList)
-					.put(Long.class, AbstractStructReader::getLongList)
-					.put(String.class, AbstractStructReader::getStringList)
-					.put(Double.class, AbstractStructReader::getDoubleList)
-					.put(Timestamp.class, AbstractStructReader::getTimestampList)
-					.put(Date.class, AbstractStructReader::getDateList)
-					.put(ByteArray.class, AbstractStructReader::getBytesList)
-					.put(Struct.class, AbstractStructReader::getStructList)
-					.build();
-
-	// @formatter:off
-	static final Map<Class, BiFunction<Struct, String, ?>> singleItemReadMethodMapping =
-			new ImmutableMap.Builder<Class, BiFunction<Struct, String, ?>>()
-					// @formatter:on
-					.put(Boolean.class, AbstractStructReader::getBoolean)
-					.put(Long.class, AbstractStructReader::getLong)
-					.put(String.class, AbstractStructReader::getString)
-					.put(Double.class, AbstractStructReader::getDouble)
-					.put(Timestamp.class, AbstractStructReader::getTimestamp)
-					.put(Date.class, AbstractStructReader::getDate)
-					.put(ByteArray.class, AbstractStructReader::getBytes)
-					.put(double[].class, AbstractStructReader::getDoubleArray)
-					.put(long[].class, AbstractStructReader::getLongArray)
-					.put(boolean[].class, AbstractStructReader::getBooleanArray).build();
-
 	private final SpannerMappingContext spannerMappingContext;
 
-	private final SpannerTypeMapper spannerTypeMapper;
+	private EntityInstantiators instantiators;
 
 	MappingSpannerReadConverter(
 			SpannerMappingContext spannerMappingContext,
 			CustomConversions customConversions) {
 		super(customConversions, null);
 		this.spannerMappingContext = spannerMappingContext;
-		this.spannerTypeMapper = new SpannerTypeMapper();
-	}
 
-	private static <R> R instantiate(Class<R> type) {
-		R object;
-		try {
-			Constructor<R> constructor = type.getDeclaredConstructor();
-			constructor.setAccessible(true);
-			object = constructor.newInstance();
-		}
-		catch (ReflectiveOperationException e) {
-			throw new SpannerDataException(
-					"Unable to create a new instance of entity using default constructor.",
-					e);
-		}
-		return object;
+		this.instantiators = new EntityInstantiators();
```

Comment:
EntityInstantiators class has many constructors it seems. We should take this in the constructor to account for the potentially different EntityInstantiator instances that the user could provide.

Reply:
yeah, the MongoDb version has [setters defined in the abstract class ](https://github.com/spring-projects/spring-data-mongodb/blob/master/spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/convert/AbstractMongoConverter.java) of the converter. 
@mp911de can you weigh in on the preferred approach around making instantiators configurable?